contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,860 @@
+                deadline: block.timestamp
+            })
+        );
+
+        (uint256 amountWETHCollected, uint256 amountOETHbCollected) = positionManager.collect(
+            INonfungiblePositionManager.CollectParams({
+                tokenId: tokenId,
+                recipient: address(this),
+                amount0Max: type(uint128).max, // defaults to all tokens owed
+                amount1Max: type(uint128).max  // defaults to all tokens owed
+            })
+        );
+
+        updateUnderlyingAssets();
+
+        emit LiquidityRemoved(
+            withdrawLiquidityShare,
+            amountWETH, //removedWETHbAmount
+            amountOETHb, //removedOETHbAmount
+            amountWETHCollected,
+            amountOETHbCollected,
+            underlyingAssets
+        );
+    }
+
+    /**
+     * @dev Perform a swap so that after the swap the ticker has the desired WETH to OETHb token share.
+     */
+    function _swapToDesiredPosition(uint256 _amountToSwap, uint256 _minTokenReceived, bool _swapWETH) internal {
+        IERC20 tokenToSwap = IERC20(_swapWETH ? WETH : OETHb);
+        uint256 balance = tokenToSwap.balanceOf(address(this));
+
+        if(balance < _amountToSwap) {
+            // if swapping OETHb
+            if (!_swapWETH) {
+               uint256 mintForSwap = _amountToSwap - balance;
+               IVault(vaultAddress).mintForStrategy(mintForSwap);
+            } else {
+                revert NotEnoughWethForSwap(balance, _amountToSwap);
+            }
+        }
+
+        // emit an event so it is easier to find correct values off-chain
+        (uint256 WETHPositionBalance, uint256 OETHbPositionBalance) = getPositionPrincipal();
+        emit PrincipalPositionBeforeSwap(WETHPositionBalance, OETHbPositionBalance);
+
+        // Swap it
+        uint256 amountReceived = swapRouter.exactInputSingle(
+            ISwapRouter.ExactInputSingleParams({
+                tokenIn: _swapWETH ? WETH : OETHb,
+                tokenOut: _swapWETH ? OETHb : WETH,
+                tickSpacing: tickSpacing, // set to 1
+                recipient: address(this),
+                deadline: block.timestamp,
+                amountIn: _amountToSwap,
+                amountOutMinimum: _minTokenReceived, // slippage check
+                // just a rough sanity check that we are within 0 -> 1 tick
+                // a more fine check is performed in _checkLiquidityWithinExpectedShare
+                sqrtPriceLimitX96: _swapWETH ? sqrtRatioX96Tick0 : sqrtRatioX96Tick1
+            })
+        );
+
+        // emit an event so it is easier to find correct values off-chain
+        (WETHPositionBalance, OETHbPositionBalance) = getPositionPrincipal();
+        emit PrincipalPositionAfterSwap(WETHPositionBalance, OETHbPositionBalance);
+    }
+
+    /**
+     * @dev Perform a swap so that after the swap the ticker has the desired WETH to OETHb token share.
+     */
+    function _addLiquidity() internal {
```

Comment:
Are we sure this method can't leave WETH on the contract?

Reply:
I will rewrite this function (according to recent Discussion on Discord) and add tests to confirm  no to little WETH is left on the contract